tests/nested/image-build/task.yaml

```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
 
 systems: [ubuntu-16.04-64]
 
-environment:
-    SSH_PORT: 8022
-    USER: user1
-    PASSWORD: ubuntu
-    SPREAD_EXTERNAL_ADDRESS: "localhost:8022"
-    WORK_DIR: /tmp/work-dir
-    TAP_DEVICE: tap0
-    VM_UNIT_COMMAND_SUFFIX: "-m 1024 -nographic -net nic,model=virtio -net user,hostfwd=tcp::${SSH_PORT}-:22 -drive file=${WORK_DIR}/ubuntu-core.img,if=virtio,cache=none -drive file=${PWD}/assertions.disk,if=virtio,cache=none"
-
 prepare: |
-    . $TESTSLIB/nested.sh
-
     # determine arch related vars
-    set_vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
 
     # create ubuntu-core image
-    mkdir -p $WORK_DIR
-    snap install --devmode --edge ubuntu-image
-    /snap/bin/ubuntu-image --image-size 3G $TESTSLIB/assertions/$model_file --channel $CORE_CHANNEL --output ubuntu-core.img
-    mv ubuntu-core.img $WORK_DIR
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
+    /snap/bin/ubuntu-image --image-size 3G $TESTSLIB/assertions/nested-${NESTED_ARCH}.model --channel $CORE_CHANNEL --output ubuntu-core.img
+    mv ubuntu-core.img /tmp/work-dir
 
     # create assertion block device
     genisoimage -volid cidata -joliet -rock -o assertions.disk $TESTSLIB/assertions/auto-import.assert
 
-    #setup_bridge $TAP_DEVICE
-
     . $TESTSLIB/systemd.sh
-    systemd_create_and_start_unit nested-vm "$vm_unit_command"
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 1024 -nographic -net nic,model=virtio -net user,hostfwd=tcp::8022-:22 -drive file=/tmp/work-dir/ubuntu-core.img,if=virtio,cache=none -drive file=${PWD}/assertions.disk,if=virtio,cache=none"
 
 restore: |
     snap remove ubuntu-image
     . $TESTSLIB/systemd.sh
     systemd_stop_and_destroy_unit nested-vm
-    rm -rf $WORK_DIR
-
-    #. $TESTSLIB/nested.sh
-    #teardown_bridge $TAP_DEVICE
+    rm -rf /tmp/work-dir
 
 execute: |
-    . $TESTSLIB/nested.sh
+    execute_remote(){
+        sshpass -p ubuntu ssh -p 8022 -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
+    }
+
+    prepare_ssh(){
```

Comment:
These two functions would be better off as comments on top of their logic instead. Using them only once and even defining them in the opposite order than they are called below seems to obfuscate what is going on unnecessarily.